### ISSUE TITLE
EC2 DescribeCustomerGateways filter 'customer-gateway-id' not implemented

### ISSUE THREAD

**Issue by kbalk** [spulec/moto#4104]
It appears the filters "tag" and "tag-key" are the only two filters supported for the EC2 API [describe_customer_gateways()](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/ec2.html#EC2.Client.describe_customer_gateways).  

Would you consider adding support for the "customer-gateway-id" filter?

**Comment by kbalk** [comment]
I think this is fixed with the transit gateway changes for 2.2.2.  I created the following test and it passed:

```
import boto3 
import pytest
from moto import mock_ec2

@mock_ec2
def test_describe_customer_gateways():
    client = boto3.client("ec2", region_name="us-east-1")
    
    customer_gateway = client.create_customer_gateway(
        Type="ipsec.1", PublicIp="205.251.242.54", BgpAsn=65534,
    ).get("CustomerGateway", {})
    gateway_id = customer_gateway["CustomerGatewayId"]
    
    response = client.describe_customer_gateways(
         Filters=[{"Name": "customer-gateway-id", "Values": [gateway_id]}],
    )
    assert response["CustomerGateways"][0]["CustomerGatewayId"] == gateway_id
```

**Comment by bblommers** [comment]
Oh, nice! That's good to hear - thanks for letting us know. Will close this then - feel free to reopen if some functionality is still missing here.
